components/admin_console/manage_roles_modal/manage_roles_modal.tsx

```diff
@@ -15,16 +17,17 @@ import {trackEvent} from 'actions/diagnostics_actions.jsx';
 import FormattedMarkdownMessage from 'components/formatted_markdown_message.jsx';
 import BotBadge from 'components/widgets/badges/bot_badge';
 import Avatar from 'components/widgets/users/avatar';
+import {isSuccess} from 'types/actions';
 
-type Props = {
+export type Props = {
     show: boolean;
     user?: UserProfile;
     userAccessTokensEnabled: boolean;
 
     // defining custom function type instead of using React.MouseEventHandler
     // to make the event optional
     onModalDismissed: (e?: React.MouseEvent<HTMLButtonElement>) => void;
-    actions: { updateUserRoles: Function };
+    actions: { updateUserRoles: (userId: string, roles: string) => ActionFunc}; // TODO add generics to ActionFunc
```

Comment:
This type should be something along the lines of `() => Promise<Something>` since `updateUserRoles` has been wrapped by `mapDispatchToProps`. If that typing isn't working out right, perhaps `mapDispatchToProps`'s type could be changed to take `dispatch: DispatchFunc` since that's the weird, overridden one we're using in mattermost-redux

Reply:
Agreed. Updated ✓